src/System.IO/tests/BufferedStream/BufferedStream.FakeStream.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+
+namespace System.IO.Tests
+{
+    /// <summary>
+    /// Provides a simple method of testing around a Stream with specific read/write/seek privileges.
+    /// </summary>
+    public class FakeStream : Stream
```

Comment:
Might be more reusable in the future if the file was moved to live in https://github.com/dotnet/corefx/tree/master/src/Common/tests/System/IO and with a more descriptive name for the type.